File: examples/src/main/scala/BasicExample.scala
import com.amazon.deequ.checks.CheckStatus
import com.amazonaws.services.glue.{DynamicFrame, GlueContext}
import com.amazonaws.services.glue.log.GlueLogger
import com.amazonaws.services.glue.util.{GlueArgParser, Job}
import dqsuite.DQSuiteContextBuilder
import org.apache.spark.SparkContext
import org.apache.spark.sql.{functions => F, types => T}

import scala.jdk.CollectionConverters.mapAsJavaMapConverter

object BasicExample {
  private val sparkContext: SparkContext = new SparkContext()
  private val glueContext: GlueContext = new GlueContext(sparkContext)
  private val spark = glueContext.getSparkSession
  private val logger = new GlueLogger()

  def main(sysArgs: Array[String]): Unit = {
    // * config_path: Path to the data quality configuration file. Can be a local file or a S3 URI.
    // * input_file_path: Path to the input file. Can be a local file or a S3 URI.
    val args = GlueArgParser.getResolvedOptions(sysArgs, Seq("JOB_NAME", "input_file_path", "config_path").toArray)
    Job.init(args("JOB_NAME"), glueContext, args.asJava)

    // Configure DQSuite
    val dqsContext = DQSuiteContextBuilder.builder
      .withConfigPath(args("config_path"))
      .withResultPath("./out/results")
      .withMetricsPath("./out/metrics")
      .withSparkSession(spark)
      .build

    val dsContext = dqsContext.withDataset("sales")

    // Load data
    val dfRaw = spark.read
      .option("header", "true")
      .option("inferSchema", "false")
      .csv(args("input_file_path"))
      .withColumn("date", F.col("date").cast(T.StringType))

    // Run schema check
    val schemaCheckResult = dsContext.schemaCheck(dfRaw)
    logger.info(
      s"Schema check finished. Found ${schemaCheckResult.numInvalidRows} invalid rows" +
        s" and ${schemaCheckResult.numValidRows} valid rows")

    // Use valid rows
    val df = schemaCheckResult.validRows

    // Run profiler and validator
    val profilingResult = dsContext.profile(df)
    logger.info(s"Profiling finished. Used ${profilingResult.numRecordsUsedForProfiling} for profiling")

    val validationResult = dsContext.validate(df)
    validationResult.status match {
      case CheckStatus.Success => logger.info("Validation succeeded")
      case CheckStatus.Warning => logger.warn("Validation succeeded with warnings")
      case CheckStatus.Error   => throw new RuntimeException("Validation failed")
    }

    val dyf = DynamicFrame(df, glueContext)
    dyf.printSchema()
  }
}

File: python/examples/basic_example.py
from pyspark.sql import SparkSession, types as T, functions as F

from dqsuite import DQSuiteContextBuilder

config_path = '../examples/data/example.yml'

spark = SparkSession.builder.getOrCreate()

dqsContext = (
    DQSuiteContextBuilder.builder(spark)
    .withConfigPath(config_path)
    .withResultPath("./out/results")
    .withMetricsPath("./out/metrics")
    .build()
)
dsContext = dqsContext.withDataset("sales")

df_raw = (
    spark.read
    .option("header", "true")
    .option("inferSchema", "false")
    .csv('../examples/data/iowa_liquor_sales_demo/iowa_liquor_sales_01.csv')
    .withColumn("date", F.col("date").cast(T.StringType()))
)

schemaCheckResult = dsContext.checkSchema(df_raw)
print(f"Schema check finished. Found {schemaCheckResult.numInvalidRows} invalid rows" +
      f" and {schemaCheckResult.numValidRows} valid rows")
assert int(schemaCheckResult.numInvalidRows) == 0, "Schema check failed"
df = schemaCheckResult.validRows

profilingResult = dsContext.profile(df)
print(f"Profiling finished. Used {profilingResult.numRecordsUsedForProfiling()} for profiling")

validationResult = dsContext.validate(df)
print(validationResult.status)
if validationResult.status == "Error":
    raise Exception("Validation failed")
elif validationResult.status == "Warning":
    print("Validation succeeded with warnings")
else:
    print("Validation succeeded")
print(validationResult.checkResults)